web/styles/rendered_markdown.css

```diff
@@ -657,13 +657,14 @@
 }
 
 .preview_content.rendered_markdown {
-    /*  Ensure that the first child and last child
-    don't have blank area at the top and bottom respectively. */
-    & :first-child {
+    /*  Ensure that the first child and last child (excluding emojis)
+    don't have blank area at the top and bottom respectively.
+    */
+    & :first-child:not(p .emoji) {
```

Comment:
How does this work? I would have expected that the first child would be a `p` that happens to contain a `.emoji` element, not a `p .emoji`... and the inspector agrees that this CSS applies when we have an emoji there. And yet when I test this, it does indeed seem that this change, which a priori I would expect to be a noop, does visually fix the bug we were seeing. I find it hard to see a reason that this change would help other than it happening to perturb things away from a browser bug.

@karlstolley I figure I'll mention you since you have the emoji CSS in your head now -- this is quite the puzzle to me!

Reply:
OK I have a theory for the evil thing happening here -- I think the `:first-child` selector as written here is NOT applying to just the first child of the `.rendered_markdown` block -- it's applying to everything that is a first child of its parent inside the container, regardless of whether it's the first child of the `rendered_markdown` block. If you test a sentence with several emoji in it, only the first emoji has this artifact.